public/app/stores/room/room-store.js

```diff
@@ -23,15 +23,36 @@ var _rooms = [
 ];
 
 var roomStore = Reflux.createStore({
-  listenables: [ roomActions ],
+  listenables: [roomActions],
 
   getRooms: function getRooms () {
     // We don't want callers to modify this array, thus we return a fresh copy
     return _rooms.slice();
   },
 
+  getDevice: function getDevice (id) {
+    var filteredDevices = _rooms.reduce(function concatDevices (devices, room) {
+      return devices.concat(room.devices);
+    }, []).filter(function isDevice (device) {
```

Comment:
It would be better to use `Array.some` and storing a reference because only the first entry is needed